### ISSUE TITLE
Allow commit to have image at node

### ISSUE THREAD

**Issue by patcon** [nicoespeon/gitgraph.js#181]
Great project! Just discovered it!

The thought was that if images were possible, then people could easily use branches of specially-themed commits adapt this to show how certain branches get deployed to various services.

I barely looked into the code yet, but do you think this would be a valuable feature? Thanks!

**Comment by nicoespeon** [comment]
Hi @patcon and thanks for the kind words 😄 

🤔 I see what you mean and I feel that'd be a very specific use case. Plus, I think it would not be easy to implement properly considering the current codebase.

I leave this open to discussion though. I feel there may be something we can do to allow people to customize the rendering by extending it, so it could be adapted to different use cases without cluttering the usage with multiple options.

**Comment by Norfeldt** [comment]
How was the gitgraph.js homepage made? It's beautiful!

Seems to me like its running markdown (or pure html) between the commits.

It's totally a feature, but it would be awesome to dump some markdown below the commit message - like a screenshot of diff or terminal commands.

**Comment by fabien0102** [comment]
@Nordeldt You can have access to source code there -> https://github.com/nicoespeon/gitgraph.js/tree/gh-pages?files=1 It's pure HTML 😉

For the markdown, I need to think about this, but it can be a good idea for v2 😀

**Comment by nicoespeon** [comment]
I realized this will be doable in v2. Labelling the issue accordingly 😉

**Comment by nicoespeon** [comment]
This is now doable with `@gitgraph/react` that was released. We even have a story for that:

![image](https://user-images.githubusercontent.com/1094774/52528740-6febc000-2cb3-11e9-93f4-a42810a194f5.png)

```jsx
const renderDot: CommitOptions["renderDot"] = (commit) => (
  // Svg of a saxophone
  <svg
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 71.84 75.33"
    height="30"
    width="30"
  >
    <g fill={commit.style.dot.color} stroke="white" strokeWidth="2">
      <path d="M68.91,35.38c4.08-1.15,3.81-3-.22-3.75-3.1-.7-18.24-5.75-20.71-6.74-2.15-1-4.67-.12-1,3.4,4,3.53,1.36,8.13,2.79,13.47C50.6,44.89,52.06,49,56,55.62c2.09,3.48,1.39,6.58-1.42,6.82-1.25.28-3.39-1.33-3.33-3.82h0L44.68,43.79c1.79-1.1,2.68-3,2-4.65s-2.5-2.29-4.46-1.93l-1.92-4.36a3.79,3.79,0,0,0,1.59-4.34c-.62-1.53-2.44-2.27-4.37-2L36,22.91c1.65-1.12,2.46-3,1.83-4.52a3.85,3.85,0,0,0-4.37-1.95c-.76-1.68-2.95-6.89-4.89-10.73C26.45,1.3,20.61-2,16.47,1.36c-5.09,4.24-1.46,9-6.86,12.92l2.05,5.35a18.58,18.58,0,0,0,2.54-2.12c1.93-2.14,3.28-6.46,3.28-6.46s1-4,2.2-.57c1.48,3.15,16.59,47.14,16.59,47.14a1,1,0,0,0,0,.11c.37,1.48,5.13,19,19.78,17.52,4.38-.52,6-1.1,9.14-3.83,3.49-2.71,5.75-6.08,5.91-12.62.12-4.67-6.22-12.62-5.81-17S66.71,36,68.91,35.38Z" />
      <path d="M2.25,14.53A28.46,28.46,0,0,1,0,17.28s3,4.75,9.58,3a47.72,47.72,0,0,0-1.43-5A10.94,10.94,0,0,1,2.25,14.53Z" />
    </g>
  </svg>
);

return (
  <Gitgraph>
    {(gitgraph) => {
      gitgraph
        .commit({ subject: "Initial commit" })
        .commit({ subject: "Another commit" })
        .commit({
          subject: "Do something crazy",
          renderDot,
        });

      gitgraph
        .branch("dev")
        .commit({
          subject: "Oh my god",
          renderDot,
        })
        .commit({
          subject: "This is a saxo!",
          renderDot,
        });
    }}
  </Gitgraph>
);
```

**Comment by Norfeldt** [comment]
Awesome! And that it is react is double awesome!

**Comment by Norfeldt** [comment]
I finally found the time to play around with it and got the storybook working. 

The `renderMessage` seems to be broken

```
.add("with render message (long)", () => {
    const renderMessage: CommitOptions["renderMessage"] = (
      commit,
      commitMessageX,
    ) => {
      return (
        <g
          transform={`translate(${commitMessageX - commit.x}, ${
            commit.style.dot.size
          })`}
        >
          <text fill={commit.style.dot.color}>
            {commit.hashAbbrev} - {commit.subject}
          </text>
          <foreignObject width="600" x="10">
            <p>
              My money's in that office, right? If she start giving me some
              bullshit about it ain't there, and we got to go someplace else and
              get it, I'm gonna shoot you in the head then and there. Then I'm
              gonna shoot that bitch in the kneecaps, find out where my goddamn
              money is. She gonna tell me too. Hey, look at me when I'm talking
              to you, motherfucker. You listen: we go in there, and that nigga
              Winston or anybody else is in there, you the first motherfucker to
              get shot. You understand?
            </p>
          </foreignObject>
        </g>
      );
    };
```

![screen shot 2019-02-23 at 15 58 08](https://user-images.githubusercontent.com/678806/53288026-fe684300-3783-11e9-9c56-5caa4671df3e.png)

**Comment by nicoespeon** [comment]
@Norfeldt indeed, the custom message stop showing up, but we fixed that with https://github.com/nicoespeon/gitgraph.js/pull/233.

It has been shipped in the last release (@gitgraph/react@1.1.0-0).
